### PR TITLE
Fix non-visual tests not cleaning up previous executions

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Game/Tests/CleanRunHeadlessGameHost.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    /// <summary>
+    /// A headless host which cleans up before running (removing any remnants from a previous execution).
+    /// </summary>
+    public class CleanRunHeadlessGameHost : HeadlessGameHost
+    {
+        public CleanRunHeadlessGameHost(string gameName = @"", bool bindIPC = false, bool realtime = true)
+            : base(gameName, bindIPC, realtime)
+        {
+            Storage.DeleteDirectory(string.Empty);
```
Comment (smoogipoo) [inline osu.Game/Tests/CleanRunHeadlessGameHost.cs]: This is not going to work for OsuTestCase, since that has a different Guid with every run.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Tests/CleanRunHeadlessGameHost.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game/Tests/CleanRunHeadlessGameHost.cs]: We should probably be deleting _after_ every run, similar to how the base `TestCase` does it.

Comment (peppy) [reply osu.Game/Tests/CleanRunHeadlessGameHost.cs]: the guid is passed in as gameName..

Comment (smoogipoo) [reply osu.Game/Tests/CleanRunHeadlessGameHost.cs]: Yes? That's what's going to stop it from working.

Comment (smoogipoo) [reply osu.Game/Tests/CleanRunHeadlessGameHost.cs]: screen shot 2018-02-08 at 5 27 07 pm

screen shot 2018-02-08 at 5 27 18 pm

Comment (peppy) [reply osu.Game/Tests/CleanRunHeadlessGameHost.cs]: this is only to ensure that we have a clean environment, not to guarantee cleanup.

we can experiment with adding it to disposal logic, but i found that to be unreliable.

Comment (smoogipoo) [reply osu.Game/Tests/CleanRunHeadlessGameHost.cs]: Right...

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**